internal/page/data.go

```diff
@@ -182,49 +182,63 @@ type Lpa struct {
 	// FeeType is the type of fee the user is applying for
 	FeeType FeeType
 	// Evidence is the documents uploaded by a donor to apply for non-full fees
-	Evidence Evidences
+	Evidence Evidence
 
 	HasSentApplicationUpdatedEvent        bool
 	HasSentPreviousApplicationLinkedEvent bool
 }
 
-type Evidences []Evidence
+type Evidence struct {
+	Documents []Document
+}
 
-func (es *Evidences) Delete(evidenceKey string) bool {
-	idx := slices.IndexFunc(*es, func(e Evidence) bool { return e.Key == evidenceKey })
+func (e *Evidence) Delete(documentKey string) bool {
+	idx := slices.IndexFunc(e.Documents, func(d Document) bool { return d.Key == documentKey })
 	if idx == -1 {
 		return false
 	}
 
-	*es = slices.Delete(*es, idx, idx+1)
+	e.Documents = slices.Delete(e.Documents, idx, idx+1)
 
 	return true
 }
 
-func (es *Evidences) Keys() []string {
+func (e *Evidence) Keys() []string {
 	var keys []string
 
-	for _, e := range *es {
-		keys = append(keys, e.Key)
+	for _, d := range e.Documents {
+		keys = append(keys, d.Key)
 	}
 
 	return keys
 }
 
-func (es *Evidences) GetByKey(key string) Evidence {
-	for _, e := range *es {
-		if e.Key == key {
-			return e
+func (e *Evidence) GetByDocumentKey(key string) Document {
+	for _, d := range e.Documents {
+		if d.Key == key {
+			return d
 		}
 	}
 
-	return Evidence{}
+	return Document{}
 }
 
-type Evidence struct {
-	Key      string
-	Filename string
-	Sent     time.Time
+func (e *Evidence) Update(document Document) bool {
```

Comment:
for the others like Attorneys I changed these to be called `Put` and to always work as the `bool` return was never going to be useful, which I think is the same case here, so could match them?

Reply:
This makes more sense - I was trying to push a bit of the logic for the lambda in here but I'm already guarding against creating a new document there so should be fine.